fvm/evm/debug/uploader.go

```diff
@@ -0,0 +1,60 @@
+package debug
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+type Uploader interface {
+	Upload(id string, data json.RawMessage) error
+}
+
+var _ Uploader = &GCPUploader{}
+
+type GCPUploader struct {
+	client *storage.Client
+	bucket *storage.BucketHandle
+}
```

Comment:
I think we had something similar to this in modules. maybe we can reuse that

Reply:
We do, but it's specific to execution data, there's no generic implementation, nor it really makes sense since it's quite light to implement this adding a wrapper would just introduce a new type. But if you have something specific in mind let me know.